### PR TITLE
User Sign Up Complete

### PR DESCRIPTION
**Pull request by bpresser123**
@bobbilee19

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -14,8 +14,25 @@
...
+
+    <%= link_to 'Edit Profile', edit_user_registration_path, :class => 'navbar-link' %> - <%= link_to "Sign Out", destroy_user_session_path, method: :delete, :class => 'navbar-link' %>
+
+    <% else %>
+
+      <%= link_to "Sign Up", new_user_registration_path, :class => 'navbar-link' %> -
+
+      <%= link_to "Sign In", new_user_session_path, :class => 'navbar-link' %>
+
+    <% end %>
```
Comment (bobbilee19) [inline app/views/layouts/application.html.erb]: keep the spaces down in this...and when you are doing and if else make sure you indent correctly. you indented right after the else but after if make sure ur fix that.

File: spec/factories/users.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory :user do
+
```
Comment (bobbilee19) [inline spec/factories/users.rb]: remove space and specs if you aren't going to make or give a shot at making a model spec...look for rspec guides on google for model specs, try signing in a user in a spec :)

**Comment by bpresser123** [comment]
@bobbilee19 Updates complete

File: spec/controllers/welcome_controller_spec.rb [hunk 3]
```diff
@@ -1,19 +0,0 @@
-require 'rails_helper'
```
Comment (bobbilee19) [inline spec/controllers/welcome_controller_spec.rb]: why did u remove this spec?

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -1,5 +0,0 @@
-require 'rails_helper'
-
-RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (bobbilee19) [inline spec/models/user_spec.rb]: add some specs in here, test user signing up? or some things to work on ur rspec skills.

File: spec/controllers/welcome_controller_spec.rb (hunk 3 above)
Comment (bpresser123) [inline spec/controllers/welcome_controller_spec.rb]: I misinterpreted your comment on `spec/factories/users.rb`. I thought you wanted me to remove specs that I wasn't using in general.

File: spec/models/user_spec.rb (hunk 4 above)
Comment (bpresser123) [inline spec/models/user_spec.rb]: My rspec skills are rough. I didn't get much practice or clarification throughout the foundation phase :(...so I definitely want to work on that!

**Comment by bpresser123** [comment]
@bobbilee19 Added rspec files and code

File: spec/models/user_spec.rb (hunk 4 above)
Comment (bobbilee19) [inline spec/models/user_spec.rb]: yep i know but it's pretty easy to google this one and see examples of models specs...for example here are some...http://everydayrails.com/2012/03/19/testing-series-rspec-models-factory-girl.html

**Comment by bpresser123** [comment]
@bobbilee19 Added bootstrap-sass gem. Let me know when to merge or pull.

**Comment by bpresser123** [comment]
@bobbilee19 I added some code to my `user_spec` and `welcome_controller_spec` files. I read through the info you sent me and watched a few videos about rspec, factory girl, and faker. All of the info and most of the syntax makes sense but I am having trouble creating my own tests. 

A lot of Rspec is still new to me so any guidance helps...maybe one example might help me write the rest of my tests? Also, my tests in the `welcome_controller_spec` fail to return http success...not sure why...I would assume those would pass considering the routes and etc are active and work with my webpage.

File: spec/factories/users.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'faker'
+
+FactoryGirl.define do
+  factory :user do
+  	email { Faker::Internet.email }
```
Comment (bobbilee19) [inline spec/factories/users.rb]: space indent not tabs

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+RSpec.describe User, type: :model do
+
+describe User do
+  it "has a valid factory" do
+    Factory.create(:user).should be_valid
+  end
+
+  it "has a valid email address" do
+
```
Comment (bobbilee19) [inline spec/models/user_spec.rb]: do you need help with these? here's an example...
let(:user) { create(:user) }
user.email = 'some email that should be valid'
expect(user.email).to be_valid

then make another test, testing that an email isn't valid and making that test past saying is expect(something).to_not be_valid

**Pull request merged**